File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/registrar/WaiterRegistrar.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.registrar;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.component.Waiter;
import org.springframework.beans.factory.support.BeanDefinitionRegistry;
import org.springframework.beans.factory.support.RootBeanDefinition;
import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
import org.springframework.core.type.AnnotationMetadata;

public class WaiterRegistrar implements ImportBeanDefinitionRegistrar {
    
    @Override
    public void registerBeanDefinitions(AnnotationMetadata metadata, BeanDefinitionRegistry registry) {
        System.out.println("WaiterRegistrar invoke ......");
        registry.registerBeanDefinition("waiter", new RootBeanDefinition(Waiter.class));
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/registrar/WaiterRegistrar.kt
package com.linkedbear.springboot.assemble.a_module.registrar

import com.linkedbear.springboot.assemble.a_module.component.Waiter
import org.springframework.beans.factory.support.BeanDefinitionRegistry
import org.springframework.beans.factory.support.RootBeanDefinition
import org.springframework.context.annotation.ImportBeanDefinitionRegistrar
import org.springframework.core.type.AnnotationMetadata

open class WaiterRegistrar : ImportBeanDefinitionRegistrar {
    override fun registerBeanDefinitions(metaData: AnnotationMetadata, registry: BeanDefinitionRegistry) {
        println("WaiterRegistrar.registerBeanDefinitions() invoked ......")
        registry.registerBeanDefinition("waiter", RootBeanDefinition(Waiter::class.java))
    }
}